File: alumni-back/src/main/java/ec/edu/ista/springgc1/model/dto/SurveyQuestionsAnswersDTO.java
package ec.edu.ista.springgc1.model.dto;

import java.io.Serializable;
import java.util.List;

import com.fasterxml.jackson.annotation.JsonView;

import ec.edu.ista.springgc1.view.View;
import lombok.Data;
@Data
public class SurveyQuestionsAnswersDTO implements Serializable{
	@JsonView(View.Public.class)
	 private Long surveyId;
	@JsonView(View.Public.class)
	    private String surveyTitle;
	@JsonView(View.Public.class)
	    private String surveyDescription;
	@JsonView(View.Public.class)
	    private List<QuestionWithAnswersDTO> questionsWithAnswers;
}

File: alumni_app/src/app/data/model/DTO/SurveyQuestionsAnswersDTO.ts
import { QuestionWithAnswersDTO } from "./QuestionWithAnswersDTO";
export class SurveyQuestionsAnswersDTO {
    surveyId: number;
    surveyTitle: string;
    surveyDescription: string;
    questionsWithAnswers: QuestionWithAnswersDTO[];
  
    constructor(
      surveyId: number,
      surveyTitle: string,
      surveyDescription: string,
      questionsWithAnswers: QuestionWithAnswersDTO[]
    ) {
      this.surveyId = surveyId;
      this.surveyTitle = surveyTitle;
      this.surveyDescription = surveyDescription;
      this.questionsWithAnswers = questionsWithAnswers;
    }
  }